main.tf

```diff
@@ -2961,7 +2963,7 @@ resource "aws_iam_role_policy_attachment" "karpenter" {
   for_each = { for k, v in {
     AmazonEKSWorkerNodePolicy          = "${local.iam_role_policy_prefix}/AmazonEKSWorkerNodePolicy",
     AmazonEC2ContainerRegistryReadOnly = "${local.iam_role_policy_prefix}/AmazonEC2ContainerRegistryReadOnly",
-    AmazonEKS_CNI_Policy               = "${local.iam_role_policy_prefix}/AmazonEKS_CNI_Policy"
+    AmazonEKS_CNI_Policy               = data.aws_eks_cluster.this.kubernetes_network_config[0].ip_family == "ipv6" ? "arn:${local.partition}:iam::${local.account_id}:policy/AmazonEKS_CNI_IPv6_Policy" : "${local.iam_role_policy_prefix}/AmazonEKS_CNI_Policy",
```

Comment:
we'll need to think about this more - this policy technically does not exist unless its created from the EKS module. For now, users can create the policy, either on their own or from the EKS module, and then pass that into the karpenter through https://github.com/aws-ia/terraform-aws-eks-blueprints-addons/blob/1b081736b7ef10626a19be20998ead8f6032262a/main.tf#L2972C51-L2972C79